### ISSUE TITLE
Suggestions for improvements.

### ISSUE THREAD

**Issue by HarveyYan** [jonbruner/ezgan#4]
This is a really nice intro for GAN. I have some suggestions that might perhaps improve the performance or fix some error (due to api change?).

1. The batch normalization layer has a "is_training" argument, which should be set to False at the time of evaluation (e.g. when generating the images). Also I believe you need to update the moving mean and variance manually (refer to tensorflow batch normalization api page).

2. You are not reusing the generator when producing fake images. Just replace 

> generator(batch_size, z_dimensions)

 with `Gz` anywhere other than its initial definition would clear this issue.

3. Use tf.variable_scope('generator') and tf.variable_scope('discriminator') instead of passing 'reuse' argument to the model function.

3. Using transposed_conv rather than resize_images when do up-sampling might improve the generator, since it gives more details instead of mere interpolations.
